include/flang/evaluate/fold.h

```diff
@@ -96,5 +96,56 @@ std::optional<std::int64_t> ToInt64(const std::optional<A> &x) {
     return std::nullopt;
   }
 }
+
+// When an expression is a constant Logical, ToBool() extracts its value.
+template<int KIND>
+std::optional<bool> ToBool(
+    const Expr<Type<TypeCategory::Logical, KIND>> &expr) {
+  if (auto scalar{
+          GetScalarConstantValue<Type<TypeCategory::Logical, KIND>>(expr)}) {
+    return scalar->IsTrue();
+  } else {
+    return std::nullopt;
+  }
+}
+
+std::optional<bool> ToBool(const Expr<SomeInteger> &);
+std::optional<bool> ToBool(const Expr<SomeType> &);
+
+template<typename A> std::optional<bool> ToBool(const std::optional<A> &x) {
+  if (x) {
+    return ToBool(*x);
+  } else {
+    return std::nullopt;
+  }
+}
+
+// When an expression is a constant character, ToString() extracts its value.
```

Comment:
How does this template function make sense for `KIND` values other than 1?  Information can be lost if other kinds of strings are converted to the default kind.

Reply:
KIND value is associated with the type. 
Eg: constCase

KIND => type{constCase->thing->GetType()}->kind()
Value => GetStringValue(constCase->thing).value()

KIND value is not lost upon conversion of Value from "u16string/u16string/string" to "string".
KIND value is still available with the expression type.

Reply:
I'm not worried about the kind value.  I'm worried about the value of the string.  You can't convert wide-character strings to 8-bit character strings without losing information.

Reply:
I am not very good in C++. I tried different approaches to avoid converting the string from one kind to another for my requirement. I got only one solution working.
Let me know if you anyone has a better approach than the implemented one.

Reply:
Now I have one function which returns different character strings without any string conversion.